src/util.ts

```diff
@@ -209,3 +210,15 @@ export function varName(s: string): string {
   // Add _ if the string has leading numbers.
   return (s.match(/^\d+/) ? '_' : '') + alphanumericS;
 }
+
+export function logicalExpr<T>(op: LogicalOperand<T>, cb: Function): string {
+  if (isLogicalNot(op)) {
+    return '!' + logicalExpr(op.not, cb);
```

Comment:
Maybe I wasn't very clear (sorry). 

To clarify, it is not needed for combining selection predicates, but can be safer for using with normal (non-selection) filter expression.  

For example,  `filter :{not: "datum.a > 10 && datum.a<0"}` should become `!(datum.a > 10 && datum.a<0)`, not `!datum.a > 10 && datum.a<0`.    

Note that `filter :{not: "datum.a > 10 && datum.a<0"}` is literally the spec and it needs extra parentheses.  We don't need extra parentheses for `filter :{not: {and: ["datum.a > 10", "datum.a<0"]}}`